### PR TITLE
Add kind configuration file for storage tests

### PR DESCRIPTION
**Pull request by jsafrane**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:

When running storage disruptive tests in a `kind` cluster, set log level of all Kubernets components to `5`. It can be done only in a kind yaml config file, so placing the file to `test/e2e/storage/kind` and moving tests that run only in `kind` cluster here.

Currently, we have the default log level `2` in the CI job and I am not able to debug why [kubelet does not start there](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/directory/pull-kubernetes-e2e-storage-kind-disruptive/1696672291175272448).

I will update [e2e test definition](https://github.com/kubernetes/test-infra/blob/008bdba2fd3853e60f07b820a9043d081239ffb9/config/jobs/kubernetes/sig-storage/sig-storage-kind.yaml#L42) to use the yaml file when running a kind cluster once this is merged.

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by mauriciopoppe — COMMENTED** [review]
Maybe you could create kind.yaml directly in the job https://github.com/kubernetes/test-infra/blob/008bdba2fd3853e60f07b820a9043d081239ffb9/config/jobs/kubernetes/sig-storage/sig-storage-kind.yaml#L42, that way if this was for debugging purposes and you're able to see why the kubelet doesn't start then you can rollback 1 change.

File: test/e2e/storage/kind/kind.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Install the default single-node kind cluster, only with KCM and kubelet log levels set to 5.
+kind: Cluster
+apiVersion: kind.x-k8s.io/v1alpha4
+nodes:
```
Comment (aojea) [inline test/e2e/storage/kind/kind.yaml:4]: is not better to have this in test-infra repo?

https://github.com/search?q=repo%3Akubernetes%2Ftest-infra%20kind.x-k8s.io%2Fv1alpha4&type=code

**Comment by jsafrane** [comment]
/close

I no longer have the immediate need for more logs, I may try test-infra next time as suggested above.

**Pull request closed without merging**